v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/range/Boundary.java

```diff
@@ -97,21 +98,44 @@ Integer stringMaxLength() {
     return partitionColumn().stringMaxLength();
   }
 
+  @Nullable
+  BigDecimal decimalStepSize() {
+    return partitionColumn().decimalStepSize();
+  }
+
   /**
    * @return builder for {@link Boundary}.
    */
   public static <T extends Serializable> Builder<T> builder() {
     return (new AutoValue_Boundary.Builder<T>()).setSplitIndex("1").setBoundaryTypeMapper(null);
   }
 
+  /**
+   * Custom equality check that relies on Object.equals() for most types. But switches to
+   * delta/min-step based comparison for: - Floating-point types (Float, Double) - Source DB type
+   * where there is a defined precision Examples: Float(p, d) TODO: Double(p, d).
+   */
+  public boolean areValuesEqual(Object valueA, Object valueB) {
+    if (valueA instanceof Float f1 && valueB instanceof Float f2) {
```

Comment:
If this is not used outside this package, can we make it protected (and possibly `@visibleForTest`)?

Please cover cases where both are not type float or one is of type flot and other is not for a complete coverage of the lines of code here. Currently the coverage for this if is partial.